### PR TITLE
Bump nettle version to 3.6

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **nettle/3.6**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/nettle/all/conanfile.py [hunk 1]
```diff
@@ -86,5 +86,10 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.libs = ["hogweed", "nettle"]
-        self.cpp_info.includedirs.append(os.path.join("include", "nettle"))
+        self.cpp_info.components["hogweed"].libs = ["hogweed"]
+        if self.options.public_key:
+            self.cpp_info.components["hogweed"].requires.append("gmp::gmp")  # FIXME: gmp::libgmp
```
Comment (madebr) [inline recipes/nettle/all/conanfile.py:91]: Depends on https://github.com/conan-io/conan-center-index/pull/1541

Comment (madebr) [inline recipes/nettle/all/conanfile.py:91]:
```suggestion
            self.cpp_info.components["hogweed"].requires.append("gmp::libgmp")
```

**Comment by madebr** [comment]
@uilianries 
Error does not make sense.
```
/usr/bin/m4:sha1-compress-2.asm:34: cannot open `source_subfolder/x86_64/sha_ni/sha1-compress.asm': No such file or directory
```
Restarting...

**Comment by uilianries** [comment]
What about adding cmake targets in test_package/CMakeLists.txt. It would be nice to exercise the components.

File: recipes/nettle/all/conanfile.py [hunk 2]
```diff
@@ -40,6 +54,8 @@ def configure(self):
             raise ConanInvalidConfiguration("Nettle cannot be built using Visual Studio")
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
+        if tools.Version(self.version) < "3.6" and self.options.get_safe("fat") and self.settings.arch == "x86_64":
+            raise ConanInvalidConfiguration("fat support is broken on this nettle release (due to missing x86_64/sha_ni/sha1-compress.asm source")
```
Comment (madebr) [inline recipes/nettle/all/conanfile.py:58]:
```suggestion
            raise ConanInvalidConfiguration("fat support is broken on this nettle release (due to a missing x86_64/sha_ni/sha1-compress.asm source)")
```

Comment (SSE4) [reply recipes/nettle/all/conanfile.py:58]: please apply suggestion

**Pull request merged**